exareme2/__init__.py

```diff
@@ -148,3 +152,57 @@ def find_flower_algorithm_folder_paths(algorithm_folders):
 flower_algorithm_folder_paths = find_flower_algorithm_folder_paths(
     FLOWER_ALGORITHM_FOLDERS
 )
+
+
+EXAFLOW_ALGORITHM_FOLDERS_ENV_VARIABLE = "EXAFLOW_ALGORITHM_FOLDERS"
+EXAFLOW_ALGORITHM_FOLDERS = "./exareme2/algorithms/exaflow"
+if exaflow_algorithm_folders := os.getenv(EXAFLOW_ALGORITHM_FOLDERS_ENV_VARIABLE):
+    EXAFLOW_ALGORITHM_FOLDERS = exaflow_algorithm_folders
+
+
+def import_exaflow_algorithm_modules() -> Dict[str, ModuleType]:
+    # Import all algorithm modules
+    # Import all .py modules in the algorithm folder paths
+    # https://stackoverflow.com/questions/67631/how-to-import-a-module-given-the-full-path?page=1&tab=votes#tab-top
+
+    all_modules = {}
+    for algorithm_folder in EXAFLOW_ALGORITHM_FOLDERS.split(","):
+        all_module_paths = glob.glob(f"{algorithm_folder}/*.py")
+        algorithm_module_paths = [
+            module
+            for module in all_module_paths
+            if isfile(module) and not module.endswith("__init__.py")
+        ]
+        algorithm_names = [
+            basename(module_path)[:-3] for module_path in algorithm_module_paths
+        ]
+        specs = [
+            importlib.util.spec_from_file_location(algorithm_name, module_path)
+            for algorithm_name, module_path in zip(
+                algorithm_names, algorithm_module_paths
+            )
+        ]
+        modules = {
+            name: importlib.util.module_from_spec(spec)
+            for name, spec in zip(algorithm_names, specs)
+        }
+
+        # Import modules
+        [
+            spec.loader.exec_module(module)
+            for spec, module in zip(specs, modules.values())
+        ]
+
+        all_modules.update(modules)
+
+    return all_modules
```

Comment:
This code is complete duplication. Pass the algorithm folder as parameter and have only one.